### PR TITLE
fix: qbd direct post onboarding testing fixes for configuration

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
### Description
fix: qbd direct post onboarding testing fixes for configuration
## Clickup
https://app.clickup.com/t/86cwpmtwg

### CODE REVIEW

File: src/app/core/models/enum/enum.model.ts [hunk 1]
```diff
@@ -396,12 +396,14 @@ export enum QBDFyleField {
 export enum QbdDirectReimbursableExportDateType {
   LAST_SPENT_AT = 'last_spent_at',
   CURRENT_DATE = 'current_date',
+  SPENT_AT = "SPENT_AT",
```
Comment (ashwin1111) [inline src/app/core/models/enum/enum.model.ts:399]: value should spent_at not SPENT_AT

File: src/app/core/models/enum/enum.model.ts [hunk 2]
```diff
@@ -396,12 +396,14 @@ export enum QBDFyleField {
...
 }
 
 export enum QbdDirectCCCExportDateType {
   SPENT_AT = 'spent_at',
   CURRENT_DATE = 'current_date',
-  POSTED_AT = 'posted_at'
+  POSTED_AT = 'posted_at',
+  LAST_SPEND_AT = "LAST_SPEND_AT"
```
Comment (ashwin1111) [inline src/app/core/models/enum/enum.model.ts:406]: last_spend_at

File: src/app/core/services/common/skip-export.service.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ export class SkipExportService {
     return this.apiService.get(`/workspaces/${this.workspaceService.getWorkspaceId()}/admin`, {});
   }
 
-  getExpenseFilter(): Observable<ExpenseFilterResponse> {
+  getExpenseFilter(): Observable<any> {
```
Comment (ashwin1111) [inline src/app/core/services/common/skip-export.service.ts:29]: why is this updated?

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-advanced-settings/qbd-direct-advanced-settings.component.ts [hunk 4]
```diff
@@ -184,18 +190,18 @@ export class QbdDirectAdvancedSettingsComponent implements OnInit {
 
   private scheduledWatcher() {
     if (this.advancedSettingsForm.controls.exportSchedule.value) {
-        this.helper.markControllerAsRequired(this.advancedSettingsForm, 'email');
+        // This.helper.markControllerAsRequired(this.advancedSettingsForm, 'email');
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-advanced-settings/qbd-direct-advanced-settings.component.ts:193]: remove commented lines

File: src/app/shared/components/configuration/configuration-skip-export/configuration-skip-export.component.ts [hunk 5]
```diff
@@ -84,39 +84,41 @@ export class ConfigurationSkipExportComponent implements OnInit {
...
     if (conditionArray.length) {
-      if (response.results[0].is_custom) {
-        this.setCustomOperatorOptions(response.results[0].rank, response.results[0].custom_field_type);
+      if (response[0].is_custom) {
+        this.setCustomOperatorOptions(response[0].rank, response[0].custom_field_type);
       } else {
-        this.operatorFieldOptions1 = this.setDefaultOperatorOptions(response.results[0].condition);
+        this.operatorFieldOptions1 = this.setDefaultOperatorOptions(response[0].condition);
       }
-      if (response.results[1]) {
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-skip-export/configuration-skip-export.component.ts:106]: let's not change anything in generic component, things will break in other apps

File: src/app/core/services/common/skip-export.service.ts (hunk 3 above)
Comment (DhaaraniCIT) [reply src/app/core/services/common/skip-export.service.ts:29]: we get the response in different way

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-advanced-settings/qbd-direct-advanced-settings.component.ts (hunk 4 above)
Comment (DhaaraniCIT) [reply src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-advanced-settings/qbd-direct-advanced-settings.component.ts:193]: we need it in future reference

File: src/app/core/services/common/skip-export.service.ts (hunk 3 above)
Comment (DhaaraniCIT) [reply src/app/core/services/common/skip-export.service.ts:29]: can we change the formate of the response

Comment (ashwin1111) [reply src/app/core/services/common/skip-export.service.ts:29]: yes, pls talk to @Hrishabh17

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-advanced-settings/qbd-direct-advanced-settings.component.ts (hunk 4 above)
Comment (ashwin1111) [reply src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-advanced-settings/qbd-direct-advanced-settings.component.ts:193]: we can add it off later whenever required

**Pull request merged**